monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -793,11 +803,13 @@ object Task extends TaskInstances {
     *
     * Although the effects are unordered, we ensure the order of results
     * matches the order of the input sequence. Also see [[gatherUnordered]].
-    *
-    * Alias for [[zipList]].
     */
-  def gather[A](in: Seq[Task[A]]): Task[List[A]] =
-    zipList(in)
+  def gather[A, M[X] <: TraversableOnce[X]](in: M[Task[A]])
+                                           (implicit cbf: CanBuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] = {
```

Comment:
That's fine. They are different anyway - the point of not guaranteeing an order is performance. Though I'm fairly certain that I'll never use `Vector` again :-)

The question is - should we return `Seq` from `gatherUnordered`? That would leave us with the freedom to change the internal implementation. But then again `Seq` is not liked in the Scala community from what I've heard, plus `List` is fairly good if you don't care about ordering and describes precisely what it is. So for now this is fine.